malcolm/modules/ADCore/parts/hdfwriterpart.py

```diff
@@ -346,14 +346,17 @@ def on_configure(self,
                               self.write_all_nd_attributes.value)
         self.layout_filename = os.path.join(
             file_dir, "%s-layout.xml" % self.mri)
-        with open(self.layout_filename, "w") as f:
-            f.write(xml)
-        layout_filename = self.layout_filename
+        layout_filename_pv = self.layout_filename
```

Comment:
I would just do:
```python
self.layout_filename = os.path.join(
            file_dir, "%s-layout.xml" % self.mri.replace(":", "_")
```
And leave the rest of the code unchanged. I would also change the replaces in the FilePathTranslator into asserts, and put replaces in the stats plugin too.

Reply:
This changes it to replace colons for underscores in Linux, but that's no bad thing...